extensions/functions_arithmetic.yaml

```diff
@@ -362,3 +362,54 @@ aggregate_functions:
         decomposable: MANY
         intermediate: fp64?
         return: fp64?
+  - name: "product"
+    description: Product of a set of values.
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i8
+        nullability: DECLARED_OUTPUT
```

Comment:
Why not `MIRROR`? I would expect the product of zero elements to be the multiplicative identity, i.e. 1, rather than null. The only other reason I see for the result to be null is if the input has nulls, which implies `MIRROR`.

Reply:
One question, wouldn't that be applicable for sum operation?

Reply:
Same applies there, but with additive identity i.e. 0. If it's defined like you defined product here then IMO that's equally wrong and that should probably be a separate or followup issue.

Reply:
Sure, I will fix this one. Shall make a minor PR to fix the SUM operation?